### PR TITLE
[MacSandbox] Extract bundle identifier into an xcconfig

### PR DESCRIPTION
**Pull request by almiliMSFT**
This will be useful because being able to control bundle identifier and buildxl class prefix from a single place will allow us to easily build and simultaneously load multiple versions of our kext (e.g., one for LKG and one for tests)

### CODE REVIEW

File: Public/Src/Sandbox/MacOs/BundleInfo.xcconfig [hunk 1]
```diff
@@ -0,0 +1,2 @@
+BXL_BUNDLE_IDENTIFIER = com.microsoft.buildxl.sandbox
```
Comment (dannyvv) [inline Public/Src/Sandbox/MacOs/BundleInfo.xcconfig]: BUILDXL_ to be consistent

File: Public/Src/Sandbox/MacOs/projects.dsc [hunk 2]
```diff
@@ -29,7 +30,7 @@ namespace Sandbox {
         Transformer.SealSourceDirectoryOption.allDirectories);
 
     export function build(args: Args): Result {
```
Comment (dannyvv) [inline Public/Src/Sandbox/MacOs/projects.dsc]: Not sure how many of these we have, but perhaps it makes sense to put a `buildXCode` function in `Sdk.BuildXL`?

File: Public/Src/Sandbox/MacOs/Sandbox/Src/BuildXLSandboxShared.hpp [hunk 3]
```diff
@@ -259,10 +259,24 @@ inline bool HasAllFlags(const T source, const T bitMask)
 
 #pragma mark Macros and defines
 
-#define BXL_CLASS(name) com_microsoft_buildxl_ ## name
-#define BuildXLSandbox BXL_CLASS(Sandbox)
-#define kBuildXLBundleIdentifier "com.microsoft.buildxl.sandbox"
-#define kBuildXLSandboxClassName "com_microsoft_buildxl_Sandbox" // this string must match the value of BuildXLSandbox
+#ifndef BXL_BUNDLE_IDENTIFIER
```
Comment (dannyvv) [inline Public/Src/Sandbox/MacOs/Sandbox/Src/BuildXLSandboxShared.hpp]: `BUILDXL_` for consistency with windows?

File: Public/Src/Sandbox/MacOs/BundleInfo.xcconfig (hunk 1 above)
Comment (almiliMSFT) [reply Public/Src/Sandbox/MacOs/BundleInfo.xcconfig]: will rename

File: Public/Src/Sandbox/MacOs/Sandbox/Src/BuildXLSandboxShared.hpp (hunk 3 above)
Comment (almiliMSFT) [reply Public/Src/Sandbox/MacOs/Sandbox/Src/BuildXLSandboxShared.hpp]: will rename

File: Public/Src/Sandbox/MacOs/projects.dsc (hunk 2 above)
Comment (almiliMSFT) [reply Public/Src/Sandbox/MacOs/projects.dsc]: currently it's just this one, so I'd say we do this once it becomes more pressing

**Pull request merged**